varinfo/variable.py

```diff
@@ -235,6 +235,25 @@ def is_temporal(self) -> bool:
         """
         return ' since ' in self.attributes.get('units', '')
 
+    def is_science(self) -> bool:
+        """ Determine if a variable is a science variable.
+            A science variable is classified as a variable that is not a
+            geographic, temporal, or a projected spatial dimension variable,
+            or if it does not have a coordinate or grid mapping reference
+            attribute.
+
+        """
+        if (self.is_geographic()
+            or self.is_temporal()
+                or self.is_projection_x_or_y()):
+            return False
+
+        if (self.references.get('coordinates')
+                or self.references.get('grid_mapping')) is not None:
```

Comment:
I think this conditional is effectively doing what you want it to, but probably not the way you expect. I think what you probably mean is:

```
if (
    self.references.get('coordinates') is not None
    of self.references.get('grid_mapping') is not None
):
    ...
```

What's happening now is the parentheses are grabbing the first "truthy" value and then comparing that to `None`. The way the `references` dictionary is defined, I think this won't be a problem, but it seems cleaner to do the direct comparison to `None` for both.